### PR TITLE
Add comprehensive test coverage for TopKLM model file (addresses #3690)

### PR DESCRIPTION
**Pull request by crocmons**
## Overview

Closes #3690

## Description

This PR adds comprehensive test coverage for `shap/models/_topk_lm.py` as part of the effort to increase overall test coverage (issue #3690). The new test file `tests/models/test_topk_lm.py` includes:

- **31 test functions** covering all major functionality of the TopKLM class
- **Complete initialization testing** for both PyTorch and TensorFlow models
- **Core functionality testing** including `__call__` method, caching, output names, and logodds computation
- **Input processing validation** for token generation and vocabulary bounds
- **Language model logits testing** for both single and batch inputs
- **Error handling verification** for invalid model types and unsupported classes
- **Serialization/deserialization testing** for save/load functionality
- **Integration testing** for batch size consistency, device handling, and result reproducibility

### Key Features:
- Tests use real tiny HuggingFace models (`hf-internal-testing/tiny-random-gpt2`) for authentic testing
- Comprehensive docstrings with clear descriptions and parameter documentation
- Coverage for previously untested code paths and bug fixes
- Proper error handling and edge case validation

This significantly improves test coverage for the TopKLM model class, making it safer to refactor and maintain.

## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [x] Unit tests added (if fixing a bug or adding a new feature)

This contribution directly addresses the test coverage goals outlined in issue #3690 and provides a solid foundation for future development of the TopKLM class.

### CODE REVIEW

**Comment by EdSheeranCoding** [comment]
Hiya!

I had actually finished fixing this and was going to make my own PR now.

Could you drop a quick comment next time to avoid duplicating effort?
would be great to coordinate on future work :)

Ciao!

**Comment by crocmons** [comment]
> Hiya!
> 
> I had actually finished fixing this and was going to make my own PR now.
> 
> Could you drop a quick comment next time to avoid duplicating effort? would be great to coordinate on future work :)
> 
> Ciao!

Actually, I raised this PR on April 10 before your comment  @EdSheeranCoding  And yeah, sure, I'll give a comment on that issue #3690 for raising other PRs. 😀 Thanks for the reminder.

File: tests/models/test_topk_lm.py [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+
+
+TEST_SENTENCES = np.array(["Hello world", "Test sentence", "Another test"])
+MASKED_SENTENCES = np.array(["Hello world", "Test sentence", "Another test"])
+BATCH_MASKED = np.array(["Test 1", "Test 2", "Test 3", "Test 4", "Test 5"])
+BATCH_X = np.array(["Test 1", "Test 2", "Test 3", "Test 4", "Test 5"])
+
+
+@pytest.fixture(scope="module")
+def pt_assets():
```
Comment (CloseChoice) [inline tests/models/test_topk_lm.py]: we can live with these failures, but I want to see them explicitly. Otherwise they might simply get lost and nobody ever cares

File: tests/models/test_topk_lm.py [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+        tok = transformers.AutoTokenizer.from_pretrained(HF_MODEL_NAME)
+    except requests.exceptions.RequestException:
+        pytest.xfail("Connection error")
+    tok.pad_token = None
+    model, _ = pt_assets
+    topk = shap.models.TopKLM(model, tok, device="cpu")
+    assert topk.tokenizer.pad_token == tok.eos_token
+
+
+@pytest.mark.skipif(
```
Comment (CloseChoice) [inline tests/models/test_topk_lm.py]: have you tested this on macos?

File: tests/models/test_topk_lm.py [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+
+    with pytest.raises(NotImplementedError, match="not supported"):
+        topk.get_lm_logits(np.array(["test"]))
+
+
+@pytest.mark.skipif(
+    platform.system() == "Darwin",
+    reason="Skipping on macOS — TF segfault risk (GH #4075)",
+)
+def test_get_lm_logits_tf_device_none(tf_assets):
```
Comment (CloseChoice) [inline tests/models/test_topk_lm.py]: can you pls get rid of this test? I don't see much value in this

File: tests/models/test_topk_lm.py [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+    with (
+        patch("transformers.MODEL_FOR_CAUSAL_LM_MAPPING") as mock_map,
+        patch("tensorflow.device", side_effect=RuntimeError("bad device")),
+    ):
+        from transformers import TFGPT2LMHeadModel
+
+        mock_map.values.return_value = list(mock_map.values.return_value) + [TFGPT2LMHeadModel]
+
+        try:
+            topk.get_lm_logits(np.array(["Hello"]))
```
Comment (CloseChoice) [inline tests/models/test_topk_lm.py]: this does not test anything.

**Review by CloseChoice — CHANGES_REQUESTED** [review]
Pls think through the tests again, I haven't read all but some look sketchy and not really adding value. Can we pls get a minimal set of tests that goes to the core of what topklm is doing?

**Comment by crocmons** [comment]
> Pls think through the tests again, I haven't read all but some look sketchy and not really adding value. Can we pls get a minimal set of tests that goes to the core of what topklm is doing?

@CloseChoice, sure, in this PR I added lots of mock test cases, some of them needed, and some of them are not necessary for now, but soon I'll update it and add a minimal set of tests that go to the core of what topklm functionality is currently doing. Thanks for the feedback.

File: tests/models/test_topk_lm.py (hunk 2 above)
Comment (crocmons) [reply tests/models/test_topk_lm.py]: No, but I have tested this on Windows.

File: tests/models/test_topk_lm.py (hunk 3 above)
Comment (crocmons) [reply tests/models/test_topk_lm.py]: Ok, sure. I had removed all the mock test cases

File: tests/models/test_topk_lm.py (hunk 4 above)
Comment (crocmons) [reply tests/models/test_topk_lm.py]: ok removed
